solana/programs/ntt-quoter/src/processor/request_relay.rs

```diff
@@ -0,0 +1,138 @@
+    #[account(
+    owner = example_native_token_transfers::ID,
+    constraint = outbox_item.released.get(WORMHOLE_TRANSCEIVER_INDEX),
+  )]
+    pub outbox_item: Account<'info, OutboxItem>,
+
+    //TODO should this be init_if_needed?
+    //     are there any security considerations here?
+    //     init_if_needed would allow for multiple requests, even after it was already delivered
+    //     right now requesting a relay for a request is permissionless which would allow
+    //       a third party to initiate a relay without gas dropoff for the user (though they
+    //       can just use connect and get a gas dropoff to the same wallet that way)
+    #[account(
+    init,
+    payer = payer,
+    space = 8 + RelayRequest::INIT_SPACE,
+    seeds = [RelayRequest::SEED_PREFIX, outbox_item.key().as_ref()],
+    bump,
+  )]
+    pub relay_request: Account<'info, RelayRequest>,
+
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(AnchorSerialize, AnchorDeserialize)]
+pub struct RequestRelayArgs {
+    pub gas_dropoff: u64, //NativeAmount,
+    pub max_fee: u64,     //SolAmount,
+}
+
+const GWEI: u64 = u64::pow(10, 9);
+
+fn mul_div(scalar: u64, numerator: u64, denominator: u64) -> u64 {
+    ((scalar as u128) * (numerator as u128) / (denominator as u128))
+        .try_into()
+        .unwrap()
+}
+
+//TODO built-in u128 division likely still wastes a ton of compute units
+//     might be more efficient to use f64 or ruint crate
+pub fn request_relay(ctx: Context<RequestRelay>, args: RequestRelayArgs) -> Result<()> {
+    let accs = ctx.accounts;
+
+    require_gt!(
+        args.gas_dropoff,
+        accs.registered_chain.max_gas_dropoff,
+        NttQuoterError::ExceedsMaxGasDropoff
+    );
+
+    let total_fee_in_lamports_without_rent = {
+        let target_native_in_gwei = args.gas_dropoff
+            + if accs.registered_chain.gas_price > 0 {
+                //gas_price[wei, 18 decimals] * EVM_GAS_COST / GWEI = target_native[gwei, 9 decimals]
+                mul_div(accs.registered_chain.gas_price, EVM_GAS_COST, GWEI)
+            } else {
+                0
+            };
+
+        //usd/target_native[usd, 6 decimals] * target_native[gwei, 9 decimals] = usd[usd, 6 decimals]
+        let target_native_in_usd = mul_div(
+            accs.registered_chain.native_price,
+            target_native_in_gwei,
+            GWEI,
+        );
+
+        let total_in_usd = target_native_in_usd + accs.registered_chain.base_price;
+
+        if total_in_usd > 0 {
+            //total_fee[sol, 9 decimals] = total_usd[usd, 6 decimals] / (sol_price[usd, 6 decimals]
+            mul_div(total_in_usd, LAMPORTS_PER_SOL, accs.instance.sol_price)
+        } else {
+            0
+        }
+    };
+
+    let rent_in_lamports = sysvar::rent::Rent::get()?.minimum_balance(8 + RelayRequest::INIT_SPACE);
+    let total_fee_in_lamports = total_fee_in_lamports_without_rent + rent_in_lamports;
```

Comment:
Currently the user is charged for the relay (`total_fee_in_lamports_without_rent`) AND rent. The relayer will claim both, so in addition to getting paid for the relay, it gets the rent lamports too. Wouldn't the right amount to charge the user be
(if we rename `total_fee_in_lamports_without_rent` to `relay_fee`)

`relay_fee.saturating_sub(rent_in_lamports)` ?

In other words, the rent amount should subsidise the relay cost, as the relayer will be able to claim it anyway.

OTOH the account in question is 8 bytes anyway, so there's a case to be made for ignoring it altogether in these calculations.

Reply:
yes, that was actually the intention, but I screwed it up. :grimacing:

Reply:
[I already had the comment in here in an early version](https://github.com/nonergodic/ntt-solana-relayer/blob/6ca388a9bd5db78345a0a439f6f2fcee28b72a7a/programs/ntt-relayer/src/processor/request_relay.rs#L111) but after trying to wrap things up (too) quickly last night I was like "hey, no need to sub if I just don't add it in!" which was a complete brain-fart...